Subject: Linux-RT 2.6.24-rc2-rt1
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
Subject: i386/x8664: cleanup the shared hpet code

Remove hpet_readl/writel from vsyscall.h, where it does not belong
anyway. Use the hpet code itself.

Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Ingo Molnar <mingo@elte.hu>

---
 arch/i386/kernel/hpet.c       |    6 +-----
 include/asm-i386/hpet.h       |    2 ++
 include/asm-x86_64/vsyscall.h |    3 ---
 3 files changed, 3 insertions(+), 8 deletions(-)

Index: linux-2.6.23-rc4/include/asm-i386/hpet.h
===================================================================
--- linux-2.6.23-rc4.orig/include/asm-i386/hpet.h	2007-08-31 21:51:18.000000000 +0200
+++ linux-2.6.23-rc4/include/asm-i386/hpet.h	2007-08-31 21:51:23.000000000 +0200
@@ -66,6 +66,7 @@
 extern unsigned long hpet_address;
 extern int is_hpet_enabled(void);
 extern int hpet_enable(void);
+extern unsigned long hpet_readl(unsigned long a);
 
 #ifdef CONFIG_HPET_EMULATE_RTC
 
@@ -85,6 +86,7 @@ extern irqreturn_t hpet_rtc_interrupt(in
 #else
 
 static inline int hpet_enable(void) { return 0; }
+static inline unsigned long hpet_readl(unsigned long a) { return 0; }
 
 #endif /* CONFIG_HPET_TIMER */
 #endif /* _I386_HPET_H */
Index: linux-2.6.23-rc4/include/asm-x86_64/vsyscall.h
===================================================================
--- linux-2.6.23-rc4.orig/include/asm-x86_64/vsyscall.h	2007-08-31 21:51:18.000000000 +0200
+++ linux-2.6.23-rc4/include/asm-x86_64/vsyscall.h	2007-08-31 21:51:23.000000000 +0200
@@ -29,9 +29,6 @@ enum vsyscall_num {
 #define VGETCPU_RDTSCP	1
 #define VGETCPU_LSL	2
 
-#define hpet_readl(a)           readl((const void __iomem *)fix_to_virt(FIX_HPET_BASE) + a)
-#define hpet_writel(d,a)        writel(d, (void __iomem *)fix_to_virt(FIX_HPET_BASE) + a)
-
 extern int __vgetcpu_mode;
 extern volatile unsigned long __jiffies;
 
Index: linux-2.6.23-rc4/arch/i386/kernel/hpet.c
===================================================================
--- linux-2.6.23-rc4.orig/arch/i386/kernel/hpet.c	2007-08-31 21:51:21.000000000 +0200
+++ linux-2.6.23-rc4/arch/i386/kernel/hpet.c	2007-08-31 21:51:23.000000000 +0200
@@ -25,11 +25,7 @@
 unsigned long hpet_address;
 static void __iomem *hpet_virt_address;
 
-/* Temporary hack. Cleanup after x86_64 clock events conversion */
-#undef hpet_readl
-#undef hpet_writel
-
-static inline unsigned long hpet_readl(unsigned long a)
+unsigned long hpet_readl(unsigned long a)
 {
 	return readl(hpet_virt_address + a);
 }
